Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +59,139 @@ class TransactionHistoryViewController: UIViewController {
     
     override func viewDidAppear(_ animated: Bool) {
         if !AppDelegate.walletLoader.isSynced {
-            print(" wallet not synced on history")
+            print("wallet not synced on history")
             return
         }
-        self.visible = true
-        if (self.FromMenu){
-            refreshControl.showLoader(in: self.tableView)
-            loadTransactions()
-            FromMenu = true
-        }
+        self.refreshControl.showLoader(in: self.tableView)
+        loadTransactions()
     }
     
-    override func viewDidDisappear(_ animated: Bool) {
-        super.viewDidDisappear(animated)
-        self.visible = false
-        dismiss(animated: true, completion: nil)
+    func initFilterBtn() {
+        self.btnFilter.initMenu(self.filterMenu) { [weak self] index, value in
+            self?.applyTxFilter(currentFilter: self!.filters[index])
+        }
     }
     
-    @objc func handleRefresh(_ refreshControl: UIRefreshControl) {
-        self.loadTransactions()
+    func applyTxFilter(currentFilter: Int) {
+        refreshControl.showLoader(in: self.tableView)
+        filteredItems.removeAll()
+        
+        switch currentFilter{
+        case 1:
+            self.filterActive = true
+            self.filteredItems = self.transactions.filter{$0.Direction == 0 && $0.Type == GlobalConstants.Strings.REGULAR}
```

Comment:
Use the constants declared in dcrlibwallet instead of the hard-coded direction integers and the swift string constants: `DcrlibwalletTxDirection*` and `DcrlibwalletTxType*`